diff -uNr wine-5.1/dlls/dbghelp/macho_module.c wine-5.1.mod/dlls/dbghelp/macho_module.c
--- wine-5.1/dlls/dbghelp/macho_module.c	2020-02-03 02:55:00.000000000 +0800
+++ wine-5.1.mod/dlls/dbghelp/macho_module.c	2022-08-11 12:33:34.801987703 +0800
@@ -1411,13 +1411,13 @@
             struct nlist nl[2];
             memset(nl, 0, sizeof(nl));
             nl[0].n_un.n_name = (char*)"_dyld_all_image_infos";
-            if (!nlist("/usr/lib/dyld", nl))
+            if (!nlist("@TERMUX_PREFIX@/lib/dyld", nl))
                 dyld_all_image_infos_addr = (void*)nl[0].n_value;
         }
 
         if (dyld_all_image_infos_addr)
         {
-            TRACE("got dyld_image_info_address %p from /usr/lib/dyld symbol table\n",
+            TRACE("got dyld_image_info_address %p from @TERMUX_PREFIX@/lib/dyld symbol table\n",
                   dyld_all_image_infos_addr);
             dyld_image_info_address = (ULONG_PTR)dyld_all_image_infos_addr;
         }
@@ -1647,7 +1647,7 @@
     {
         const char* fallback = getenv("DYLD_FALLBACK_LIBRARY_PATH");
         if (!fallback)
-            fallback = "/usr/local/lib:/lib:/usr/lib";
+            fallback = "@TERMUX_PREFIX@/lib";
         ret = macho_load_file_from_path(pcs, p, load_addr, fallback, macho_info);
     }
     if (!ret && !strchrW(filename, '/'))
diff -uNr wine-5.1/libs/wine/config.c wine-5.1.mod/libs/wine/config.c
--- wine-5.1/libs/wine/config.c	2020-02-03 02:55:00.000000000 +0800
+++ wine-5.1.mod/libs/wine/config.c	2022-08-11 12:38:01.061987601 +0800
@@ -42,8 +42,8 @@
 #endif
 #include "wine/library.h"
 
-static const char server_config_dir[] = "/.wine";        /* config dir relative to $HOME */
-static const char server_root_prefix[] = "/tmp/.wine";   /* prefix for server root dir */
+static const char server_config_dir[] = "@TERMUX_HOME@/.wine";        /* config dir relative to $HOME */
+static const char server_root_prefix[] = "@TERMUX_PREFIX@/tmp/.wine";   /* prefix for server root dir */
 static const char server_dir_prefix[] = "/server-";      /* prefix for server dir */
 
 static char *bindir;
diff -uNr wine-5.1/server/request.c wine-5.1.mod/server/request.c
--- wine-5.1/server/request.c	2020-02-03 02:55:00.000000000 +0800
+++ wine-5.1.mod/server/request.c	2022-08-11 12:41:26.971987523 +0800
@@ -662,15 +662,15 @@
 
     /* create the base directory if needed */
 
-#ifdef __ANDROID__  /* there's no /tmp dir on Android */
+#ifdef __ANDROID__  && ! defined(__TERMUX__)/* there's no /tmp dir on Android */
     len += strlen( config_dir ) + sizeof("/.wineserver");
     if (!(server_dir = malloc( len ))) fatal_error( "out of memory\n" );
     strcpy( server_dir, config_dir );
     strcat( server_dir, "/.wineserver" );
 #else
-    len += sizeof("/tmp/.wine-") + 12;
+    len += sizeof("@TERMUX_PREFIX@/tmp/.wine-") + 12;
     if (!(server_dir = malloc( len ))) fatal_error( "out of memory\n" );
-    sprintf( server_dir, "/tmp/.wine-%u", getuid() );
+    sprintf( server_dir, "@TERMUX_PREFIX@/tmp/.wine-%u", getuid() );
 #endif
     create_dir( server_dir, &st2 );
 
