diff -uNr uftrace-0.14/cmds/live.c uftrace-0.14.mod/cmds/live.c
--- uftrace-0.14/cmds/live.c	2023-06-20 13:58:52.000000000 +0800
+++ uftrace-0.14.mod/cmds/live.c	2023-06-21 20:27:12.785999889 +0800
@@ -416,7 +416,7 @@
 int command_live(int argc, char *argv[], struct uftrace_opts *opts)
 {
 #define LIVE_NAME "uftrace-live-XXXXXX"
-	char template[32] = "/tmp/" LIVE_NAME;
+	char template[32] = "@TERMUX_PREFIX@/tmp/" LIVE_NAME;
 	int fd;
 	struct sigaction sa = {
 		.sa_flags = SA_RESETHAND,
@@ -432,7 +432,7 @@
 			strcpy(template, LIVE_NAME);
 
 			if (errno != EPERM && errno != ENOENT)
-				pr_err("cannot access to /tmp");
+				pr_err("cannot access to @TERMUX_PREFIX@/tmp");
 
 			fd = mkstemp(template);
 
